Logstash/logstash_http.conf

```diff
@@ -19,52 +24,100 @@ filter {
 
 	#ruby code to convert the comma seperated roles into an array
 	ruby{
-	     code =>"
+	    code =>"
 
-		 # method to split the supplied string by comma, trim whitespace and return an array
-		 def mapStringToArray(strFieldValue)
+		# method to split the supplied string by comma, trim whitespace and return an array
+		def mapStringToArray(strFieldValue)
 
-		    #if string is not null, return array
-		    if (strFieldValue != nil)
-		        fieldArr =  strFieldValue.split(',').map(&:strip).reject(&:empty?).uniq 
-		        return fieldArr                             
-		    end     
+		#if string is not null, return array
+		if (strFieldValue != nil)
+			fieldArr =  strFieldValue.split(',').map(&:strip).reject(&:empty?).uniq 
+			return fieldArr                             
+		end     
 
-		    return [] #return empty array if string is nil
-		 end
+		return [] #return empty array if string is nil
+		end
 
-		 vrtArr = mapStringToArray(event.get('[event][payloadData][rolesCommaSeparated]'))
-		 if vrtArr.length > 0                           
-		    event.set('[event][payloadData][rolesCommaSeparated]', vrtArr) 
-		 end
+		vrtArr = mapStringToArray(event.get('[event][payloadData][rolesCommaSeparated]'))
+		if vrtArr.length > 0                           
+		event.set('[event][payloadData][rolesCommaSeparated]', vrtArr) 
+		end
+
+		event_array = []
+		event_type = event.get('[event][payloadData][eventType]')
+		idp_type = event.get('[event][payloadData][identityProviderType]')
+		auth_step_success = event.get('[event][payloadData][authStepSuccess]')
+        if (event_type == 'overall' || auth_step_success == false)
+			event_array.push('Overall')
+		end
+		if (event_type == 'step')
+			if (idp_type == 'Resident')
+				event_array.push('Resident')
+			end
+			if (idp_type == 'FEDERATED')
+				event_array.push('Federated')
+			end
+		end
+		event.set('[event][payloadData][event_type_filter]', event_array)
 	"
 	}
 
-	# debug timestamp
-	if ("" in [mtime]) {
-		date {
-    			match => [ "mtime" , "yyyy-MM-dd HH:mm:ss.SSS" ]
-    			target => "@timestamp"
-			locale => "en"
-			timezone => "Asia/Colombo"
+	# session specific
+	if [eventType] == "session" {
+
+		# user agent filter to transform the userAgent string into seperated fields
+		useragent {
+			ecs_compatibility => disabled
+			source => "[event][payloadData][userAgent]"
+			target => "userAgentDetails"
+		}
+
+	} else if [eventType] == "auth" {
+
+		#geo ip filter to transform the IP address to location
```

Comment:
fix the comment format

Reply:
Fixed with https://github.com/wso2-extensions/identity-elk-integration/pull/14